### PR TITLE
Added support in nmcli for ipv4.dns-options

### PR DESCRIPTION
**Pull request by M425**
##### SUMMARY

the nmcli command has an important option "ipv4.dns-options", This PR adds the support for this option in the nmcli  module

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

community.general.nmcli --> plugins/modules/net_tools/nmcli.py

##### ADDITIONAL INFORMATION

Simple pull request to add the feature to the existing module

### CODE REVIEW

File: plugins/modules/net_tools/nmcli.py [hunk 1]
```diff
@@ -124,6 +124,11 @@
             - A list of DNS search domains.
         elements: str
         type: list
+    dns4_options:
+        description:
+            - A list of DNS options.
+        elements: str
+        type: list
```
Comment (felixfontein) [inline plugins/modules/net_tools/nmcli.py:131]:
```suggestion
        type: list
        version_added: 4.6.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Also extending the [unit tests](https://github.com/ansible-collections/community.general/blob/main/tests/unit/plugins/modules/net_tools/test_nmcli.py) would be helpful.

Finally, does it also makes sense to add a `dns6_options` option? Or does this have no IPv6 counterpart in `nmcli` itself?

**Comment by M425** [comment]
> itself

Added support for dns6-options

**Comment by M425** [comment]
> Finally, does it also makes sense to add a `dns6_options` option? Or does this have no IPv6 counterpart in `nmcli` itself?

Added dns6-options

**Comment by M425** [comment]
> Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Also extending the [unit tests](https://github.com/ansible-collections/community.general/blob/main/tests/unit/plugins/modules/net_tools/test_nmcli.py) would be helpful.

Added changelog, not a big expert of unit test, and that file is huge to analyze, can you please help me through?

File: changelogs/fragments/4308-added-support-in-nmcli-for-ipvx-dns-options.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - nmcli - add support for ipv4.dns-options and ipv6.dns-options
```
Comment (felixfontein) [inline changelogs/fragments/4308-added-support-in-nmcli-for-ipvx-dns-options.yml:2]:
```suggestion
  - nmcli - add support for ``ipv4.dns-options`` and ``ipv6.dns-options`` (https://github.com/ansible-collections/community.general/pull/4308).
```

**Comment by felixfontein** [comment]
> > Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Also extending the [unit tests](https://github.com/ansible-collections/community.general/blob/main/tests/unit/plugins/modules/net_tools/test_nmcli.py) would be helpful.
> 
> Added changelog, not a big expert of unit test, and that file is huge to analyze, can you please help me through?

I'm not familiar with the nmcli tests either, but once you fix the remaining main issue (i.e. get rid of the merge commit) the bot should ping the module maintainers, and these should be able to help you with this.

**Comment by felixfontein** [comment]
CC @trewest @Ajpantuso @haddystuff since you contributed to this module recently.

**Comment by trewest** [comment]
Hey @M425, you can add your unit tests here: [test_nmcli.py](https://github.com/ansible-collections/community.general/blob/main/tests/unit/plugins/modules/net_tools/test_nmcli.py). 

What I would recommend, is just testing that your changes work with Ansible first, then you can translate those tests into a few unit tests and base the structure on the preexisting tests. We use ansible-test for running the tests and you can find more information about it [here](https://docs.ansible.com/ansible/devel/dev_guide/developing_collections_testing.html#testing-tools). I generally test locally using this command:

` ansible-test units -v --docker --python 3.8 tests/unit/plugins/modules/net_tools/test_nmcli.py`

**Comment by felixfontein** [comment]
@M425 could you please add tests for this new feature? Thanks.

**Comment by felixfontein** [comment]
needs_info

File: plugins/modules/net_tools/nmcli.py [hunk 3]
```diff
@@ -124,6 +124,12 @@
...
+        version_added: 4.6.0
```
Comment (felixfontein) [inline plugins/modules/net_tools/nmcli.py:132]:
```suggestion
        version_added: 5.0.0
```

File: plugins/modules/net_tools/nmcli.py [hunk 4]
```diff
@@ -184,6 +190,12 @@
             - A list of DNS search domains.
         elements: str
         type: list
+    dns6_options:
+        description:
+            - A list of DNS options.
+        elements: str
+        type: list
+        version_added: 4.6.0
```
Comment (felixfontein) [inline plugins/modules/net_tools/nmcli.py:198]:
```suggestion
        version_added: 5.0.0
```

**Comment by M425** [comment]
can't add test, but the change works fine. can somebody help me on the tests?

**Comment by russoz** [comment]
> can't add test, but the change works fine. can somebody help me on the tests?

There is a unit test file for nmcli already at: `tests/unit/plugins/modules/net_tools/test_nmcli.py`

That should be a good starting point.

**Comment by felixfontein** [comment]
Please note that in #5461 the collection repository was restructured to remove the directory tree in plugins/modules/, and the corresponding tree in tests/unit/plugins/modules/. Your PR modifies files in this directory structure, and unfortunately now has some conflicts, potentially because of this. Please **rebase** with the current `main` branch to remove the conflicts.

**Comment by felixfontein** [comment]
needs_info

**Comment by yan12125** [comment]
@M425: Hi, are you still working on this? I'm interested in this feature, and I'm willing continue on the efforts.

**Comment by yan12125** [comment]
Probably the original author is not working on this. I created a rebased and improved version at https://github.com/ansible-collections/community.general/pull/6902

**Pull request closed without merging**